x/ecocredit/msgs.go

```diff
@@ -8,15 +8,19 @@ import (
 )
 
 var (
-	_, _, _, _, _, _ sdk.Msg = &MsgCreateClass{}, &MsgCreateBatch{}, &MsgSend{},
-		&MsgRetire{}, &MsgCancel{}, &MsgSetPrecision{}
+	_, _, _, _, _ sdk.Msg = &MsgCreateClass{}, &MsgCreateBatch{}, &MsgSend{},
+		&MsgRetire{}, &MsgCancel{}
 )
 
 func (m *MsgCreateClass) ValidateBasic() error {
 	if len(m.Issuers) == 0 {
 		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "issuers cannot be empty")
 	}
 
+	if len(m.CreditType) == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "credit class must have a credit type")
```

Comment:
```suggestion
		return sdkerrors.ErrInvalidRequest.Wrap("credit class must have a credit type")
```

Saves repeating `sdkerrors`